### PR TITLE
Enable FindBugs runs in the pull request builder

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Enables FindBugs reporting on packaging stages and https://jenkins.ci.cloudbees.com/job/plugins/job/git-plugin/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,7 +45,18 @@
           <configuration>
             <excludeFilterFile>src/findbugs/excludesFilter.xml</excludeFilterFile>
             <failOnError>${maven.findbugs.failure.strict}</failOnError>
+            <xmlOutput>true</xmlOutput>
+            <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
```
Comment (KostyaSha) [inline pom.xml]: why it needed?

Comment (oleg-nenashev) [inline pom.xml]: It's required in order to parse XML reports

Comment (KostyaSha) [inline pom.xml]: Messages DB is maintained in FB plugins AFAIK. @uhafner ?

File: pom.xml [hunk 2]
```diff
@@ -45,7 +45,18 @@
...
           </configuration>
+          <executions>
+            <execution>
+              <id>run-findbugs</id>
```
Comment (KostyaSha) [inline pom.xml]: why not just call it `findbugs`? Any execution that is run is run.

File: pom.xml (hunk 1 above)
Comment (oleg-nenashev) [inline pom.xml]: These messages provide context details. E.g.: "Access to field foo is not synchronized. Synchronized 66% times"

Comment (oleg-nenashev) [inline pom.xml]: See https://jenkins.ci.cloudbees.com/job/plugins/job/git-plugin/872/findbugsResult/ for example

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: I may add another plugin with "publish-findbugs" later, for example

Comment (KostyaSha) [inline pom.xml]: run-publish-findbugs, ok

**Comment by KostyaSha** [comment]
Seems you overworked and calls git-plugin as pull request builder plugin.

**Comment by MarkEWaite** [comment]
This change will cause "mvn clean install" to fail if any findbugs warnings are still open.  Currently there are several findbugs warnings in the code.  I've been maintaining a [findbugs fixes branch](https://github.com/MarkEWaite/git-plugin/tree/master-findbugs-fixes) that resolves the warnings, but need more time to evaluate those fixes before this change is merged to master.

I'd like this change to wait until at least after the upcoming release of git plugin 2.4.0.  That should happen within the next week.

**Comment by KostyaSha** [comment]
> I'd like this change to wait until at least after the upcoming release of git plugin 2.4.0. That should happen within the next week.

+1 for delay. No reason for rush in blind FB fixing

**Comment by oleg-nenashev** [comment]
@MarkEWaite 
Agreed. I you don't mind I'll fix existing FindBugs issue in this pull request

**Comment by oleg-nenashev** [comment]
@MarkEWaite 
I've occasionally fixed the bugs in parallel (excepting serialization class conflicts). Would it be possible to merge our branches. 

I've also disable build failures by default

**Comment by oleg-nenashev** [comment]
@MarkEWaite 
Any plans regarding the integration and merge?

**Comment by MarkEWaite** [comment]
Yes, I've started including your changes into the [master-findbugs-fixes branch|https://github.com/MarkEWaite/git-plugin/tree/master-findbugs-fixes] I had started on my fork.  I prefer the changes as a series of commits, each fixing a particular type of findbugs warning.  That makes it clearer (to me) what was being corrected and simplifies revert if one of the fixes causes a problem.  It will likely be a few more weeks before I can complete that process.

**Comment by oleg-nenashev** [comment]
@MarkEWaite 
Thanks a lot for your efforts. Next time just let me know if you prefer another PRs approach. I could create separate PRs for issues and reduce your efforts.

File: src/main/java/hudson/plugins/git/GitRevisionBuildParameters.java [hunk 3]
```diff
@@ -51,22 +51,32 @@ public GitRevisionBuildParameters(boolean combineQueuedCommits) {
 	public GitRevisionBuildParameters() {
 	}
 
-	@Override
-	public Action getAction(AbstractBuild<?,?> build, TaskListener listener) {
-		BuildData data = build.getAction(BuildData.class);
-		if (data == null && Jenkins.getInstance().getPlugin("promoted-builds") != null) {
+    @Override
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitRevisionBuildParameters.java]: indentation?

File: src/main/java/hudson/plugins/git/GitStatus.java [hunk 4]
```diff
@@ -36,6 +36,7 @@
  */
 @Extension
 public class GitStatus extends AbstractModelObject implements UnprotectedRootAction {
+
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitStatus.java]: why?

**Comment by oleg-nenashev** [comment]
Has been fixed by the Parent POM update

**Pull request closed without merging**